SIP325 document proposed C and N limited leaf-on events (#332)
PecanProject/sipnet: docs/model-structure.md

There are multiple ways to configure the model structure, and not all model structures or components are listed.
Implementation in source code (sipnet.c) is annotated with references to specific publications.

> 

Author: ```suggestion
```

#### Notes on notation:

- The general approach used to define variables and subscripts is defined in [Notation](parameters.md#notation).
- Specific parameter, flux, and state definitions are documented in [Model States and Parameters](parameters.md#run-time-parameters).

> Symbols in blackboard bold, e.g. $\mathbb{X}$ indicates proposed model quantities that are documented here but

Author: ```suggestion
Symbols in blackboard bold, e.g. $\mathbb{X}$, indicate proposed model quantities that are documented here but
```

> Symbols in blackboard bold, e.g. $\mathbb{X}$ indicates proposed model quantities that are documented here but

Author: already fixed

- The general approach used to define variables and subscripts is defined in [Notation](parameters.md#notation).
- Specific parameter, flux, and state definitions are documented in [Model States and Parameters](parameters.md#run-time-parameters).

Symbols in blackboard bold, e.g. $\mathbb{X}$, indicate proposed model quantities that are documented here but

> not yet implemented.

Reviewer: I'm not sure this is worth it at this point? Leave it in if you feel strongly, of course.

Author: I do appreciate the opportunity to use some of the cool fonts, but what I feel strongly about is making sure that the docs reflect the actual implementation.

I would be okay with the alternative of adding implementation to this PR or otherwise leaving this PR open until the feature is implemented.

Reviewer 2: Would it work to write like `[\mathrm{planned}] F^C_{\text{mort},j,t} = ...`? Then nobody has to remember any extra notations

Author: @infotroph unless there is objection, lets go with 'wait to merge docs until implemented'

\begin{equation}
f_{\text{leaf,on}} =
\min(f^C_{\text{limit}}, f^N_{\text{limit}})
\label{eq:leaf_on_limiter}
\end{equation}

with $0 \le f^C_{\text{limit}} \le 1$ and $0 \le f^N_{\text{limit}} \le 1$.

Carbon available for leaf growth at leaf-on comes from a fraction of the perennial wood and coarse root biomass pools:

\begin{equation}
C_{\text{realloc}} =
f^C_{\text{realloc}} \cdot
(C_{\text{wood}} + C_{\text{coarse root}})
\label{eq:leaf_on_realloc_c}
\end{equation}

where $0 \le f^C_{\text{realloc}} \le 1$. Leaf-on reallocation uses structural perennial biomass only. It does not draw
from $C_{\text{wood,storage}}$, because the storage pool is a bookkeeping buffer for recent carbon allocation lag rather than a carbon pool available for reallocation.

If structural biomass is greater than zero, carbon is transferred to leaf carbon. 
Carbon is taken from the source pools in proportion to their size:

\begin{equation}
F^C_{\text{source,}j} =
F^C_{\text{leaf,on}}
\frac{C_j}{C_{\text{wood}} + C_{\text{coarse root}}}
\label{eq:leaf_on_source_c}
\end{equation}

\begin{equation*}
\small j \in \{\text{wood, coarse root}\}
\end{equation*}

This makes the source-pool debit auditable:

> \begin{equation}
> \sum_j F^C_{\text{source,}j} = F^C_{\text{leaf,on}}
> \label{eq:leaf_on_source_c_sum}
> \end{equation}

Reviewer: One side of this should be the negative of the other; perhaps: 

```suggestion
\begin{equation}
\sum_j F^C_{\text{source,}j} + F^C_{\text{leaf,on}} = 0
\label{eq:leaf_on_source_c_sum}
\end{equation}
```

Author: I'll defer to you, but to clarify the rationale behind stating:

$$
\text{inputs}=\text{outputs}
$$

Is that by convention in this document, fluxes are > 0, and +/- signs are used in state updates:

$$
\frac{dX}{dt}=\text{inputs}-\text{outputs}
$$

At one point I considered of indicating direction like:

$$F^C_{\text{source}\rightarrow\text{sink}}$$

But I was already spending too much time on notation and I don't know if it would clarify this case 🤯.

Reviewer: I'm fine either way, really. Your current form does seem to be in keeping with the rest of the doc.

the plant will never regrow leaf biomass.

The carbon-side limiter is:

\begin{equation}
f^C_{\text{limit}} =
\min\left(1, \frac{C_{\text{realloc}}}{F^C_{\text{demand,leaf,on}}}\right)
\label{eq:leaf_on_c_limiter}
\end{equation}

and the nitrogen-side limiter is:

\begin{equation}
f^N_{\text{limit}} =
\min\left(1, \frac{F^N_{\text{supply,leaf,on}}}{F^N_{\text{demand,leaf,on}}}\right)
\label{eq:leaf_on_n_limiter}
\end{equation}

where $F^N_{\text{supply,leaf,on}}$ includes plant N storage, mineral uptake, and fixation.

#### Leaf Off {#leaf-off}

Leaf-off events define the timing of leaf senescence. A leaf-off event transfers leaf biomass carbon out of the leaf
pool. When the litter pool is enabled, leaf litter enters the litter pool; otherwise, leaf litter is
routed to soil carbon. A fraction of nitrogen from senescing leaves is resorbed into a plant storage pool before the
rest is transferred to litter or soil. The resorption flux is:

\begin{equation}
F^N_{\text{storage,in}} =
f^N_{\text{resorb}} \cdot F^N_{\text{senescing,leaf}}
\label{eq:leaf_n_resorb}
\end{equation}

> 

Reviewer: ```suggestion

where 
\begin{equation}
F^N_{\text{senescing,leaf}} = F^C_{\text{senescing,leaf}} \cdot CN_\text{leaf}
\end{equation}
```

and $F^C_{\text{senescing,leaf}}$ is defined in my suggestion above.

\begin{equation}
F^N_{\text{litter,leaf,residual}} =
(1 - f^N_{\text{resorb}}) \cdot F^N_{\text{senescing,leaf}}
\label{eq:leaf_n_litter_residual}
\end{equation}

where $0 \le f^N_{\text{resorb}} \le 1$. The plant nitrogen storage pool balance is:

\begin{equation}
\frac{dN_{\text{plant,storage}}}{dt} =
F^N_{\text{storage,in}} - F^N_{\text{storage,use}}
\label{eq:plant_n_storage}
\end{equation}

> Stored nitrogen contributes to satisfying plant growth nitrogen demand, including leaf-on demand.

Reviewer: ```suggestion
Stored nitrogen contributes to satisfying plant growth nitrogen demand, including leaf-on demand (indicated as $F^N_{\text{storage,use}}$).
```
Or some other way to explicitly mention that F term.

Reviewer: Also, should we explicitly say "can be used for any plant N demand, not just leaf on", or is that obvious? (Maybe just an "all" before "plant growth nitrogen demand"?)

Feel free to not do anything here, I think this is just me.

The change in plant leaf carbon $(C_\text{leaf})$ over time is given by the balance of leaf production $(L)$ and leaf
litter production $(F^C_\text{litter,leaf})$.

#### Leaf On {#leaf-on}

> Leaf-on events define the timing of leaf emergence. Leaf-on requests a transfer of carbon to the leaf pool, and the
> amount transferred is limited by available carbon and nitrogen:

Author: ```suggestion
Leaf-on events define the timing of leaf emergence. 
Leaf-on requests a transfer of carbon to the leaf pool, and the
amount transferred is limited by available carbon and nitrogen.
The requested leaf-on carbon demand $F^C_{\text{demand,leaf,on}}$ is bounded by the seasonal leaf-growth amount parameter, $\Delta C_{\text{leaf}}$:
```

Reviewer: Isn't F*C_{demand,leaf,on} the same as the Delta_C param? I can't find where we define that F term.

Author: Yeah this is pretty confusing and should be re-written.

Goal is:
- F^C_leaf is realized flux of carbon to leaf.
- Delta_C_leaf (misleading because it is not realized) is the parameter that defines how much leaf biomass is added if not limited by C or N availability.

\begin{equation}
\mathbb{F}^C_{\text{leaf,on,}t} =
f_{\text{leaf,on,}t} \cdot F^C_{\text{demand,leaf,on,}t}
\label{eq:leaf_on_creation}
\end{equation}

where $0 \le f_{\text{leaf,on,}t} \le 1$.

A minimal limiter combines carbon and nitrogen constraints:

\begin{equation}
\mathbb{f}_{\text{leaf,on,}t} =
\min(f^C_{\text{limit,}t}, f^N_{\text{limit,}t})
\label{eq:leaf_on_limiter}
\end{equation}

with $0 \le f^C_{\text{limit,}t} \le 1$ and $0 \le f^N_{\text{limit,}t} \le 1$.

Carbon available for leaf growth at leaf-on comes from a fraction of the perennial wood and coarse root biomass pools:

\begin{equation}
\mathbb{C}_{\text{realloc,}t} =
\mathbb{f}^C_{\text{realloc}} \cdot
(C_{\text{wood,}t} + C_{\text{coarse root,}t})
\label{eq:leaf_on_realloc_c}
\end{equation}

where $0 \le \mathbb{f}^C_{\text{realloc}} \le 1$. Leaf-on reallocation uses structural perennial biomass only. It does not draw

> from $C_{\text{wood,storage}}$, because the storage pool is a bookkeeping buffer for recent carbon allocation lag rather than a carbon pool available for reallocation.

Reviewer: Yeah, I like this.

Note that this is making me feel more strongly about resolving whether the storage pool should be included for calcs of wood resp, wood turnover, and harvest.

Author: Actually, how about renaming the "storage" pool to something to indicate that it is an accounting tool rather than a biological representation, like `[NPP|Wood]Allocation[Lag|Buffer]`?

Reviewer: Thinking about this more, I'm actually not sure I totally agree. The storage pool is a buffer used to track carbon entering and leaving the system without any associated nitrogen. We need it to balance nitrogen, since we have fixed C:N ratios.

Thinking of it as just a allocation lag tracker doesn't feel right when that pool is negative.
When it's positive, I think I agree with your wording - it's basically a lag buffer. When it's negative, though, some carbon pool somewhere is taking a hit, we just don't reduce any pools because we are also using those pool sizes to track nitrogen, which hasn't changed.

Reviewer: That's not to say I don't favor renaming it (apologies for the double negative, it's a real failing of mine 😂)

Author: Yeah, the key point was 

> renaming the "storage" pool to something to indicate that it is an accounting tool rather than a biological representation

I'll leave naming to those more qualified!

Author: To get back to your original point

> ... resolving whether the storage pool should be included for calcs of wood resp, wood turnover, and harvest

If we aren't going to consider this a true storage pool, it should not be included in wood respiration, wood turnover, harvest, or leaf-on reallocation.

If structural biomass is greater than zero, carbon is transferred to leaf carbon. 
Carbon is taken from the source pools in proportion to their size:

\begin{equation}
\mathbb{F}^C_{\text{source,}j,t} =
F^C_{\text{leaf,on,}t}
\frac{C_{j,t}}{C_{\text{wood,}t} + C_{\text{coarse root,}t}}
\label{eq:leaf_on_source_c}
\end{equation}

\begin{equation*}
\small j \in \{\text{wood, coarse root}\}
\end{equation*}

If the source biomass is zero, realized leaf-on growth is zero. If the source biomass is zero and there are no leaves,
the plant will never regrow leaf biomass.

The carbon-side limiter is:

\begin{equation}
\mathbb{f}^C_{\text{limit,}t} =
\min\left(1, \frac{C_{\text{realloc,}t}}{F^C_{\text{demand,leaf,on,}t}}\right)
\label{eq:leaf_on_c_limiter}
\end{equation}

and the nitrogen-side limiter is:

\begin{equation}
\mathbb{f}^N_{\text{limit,}t} =
\min\left(1, \frac{F^N_{\text{supply,leaf,on,}t}}{F^N_{\text{demand,leaf,on,}t}}\right)
\label{eq:leaf_on_n_limiter}
\end{equation}

> where $F^N_{\text{supply,leaf,on,}t}$ includes plant N storage, mineral N uptake, and N fixation.

Reviewer: I've been fearing this one, as I could see it coming. This will complicate flow a fair amount. It can be done, but this is not trivial.

Author: if it is any consolation, at least we can defer mortality 🤞🏻

Reviewer: We _hope_...

#### Leaf Off {#leaf-off}

Leaf-off events define the timing of leaf senescence. A leaf-off event transfers leaf biomass carbon out of the leaf
pool. When the litter pool is enabled, leaf litter enters the litter pool; otherwise, leaf litter is
routed to soil carbon. A fraction of nitrogen from senescing leaves is resorbed into a plant storage pool before the

> rest is transferred to litter or soil. The resorption flux is:

Author: ```suggestion
Leaf-off events define the timing of leaf senescence. 
A leaf-off event transfers leaf biomass carbon out of the leaf pool. 
The leaf-off carbon transfer is controlled by the leaf fall 
fraction parameter, $f_{\text{fall}}$.

\begin{equation}
F^C_{\text{senescing,leaf}} = f_\text{fall} \cdot C_\text{leaf}
\end{equation}

When the litter pool is enabled, leaf litter enters the litter pool; 
otherwise, leaf litter is routed to soil carbon.

At leaf-off, a fraction of nitrogen from senescing leaves is resorbed 
into a plant storage pool before the rest is transferred to litter or soil. 
The resorption flux is:
```

Reviewer: This looks good, feel free to apply.

Reviewer: Except maybe put this in after line 288:

\begin{equation}
F^C_{\text{senescing,leaf}} = f_\text{fall} \cdot C_\text{leaf}
\end{equation}

The change in plant leaf carbon $(C_\text{leaf})$ over time is given by the balance of leaf production $(L)$ and leaf
litter production $(F^C_\text{litter,leaf})$.

#### Leaf On {#leaf-on}

> Leaf-on events define the timing of leaf emergence. Leaf-on requests a transfer of carbon to the leaf pool, and the

Reviewer: Should we explicitly say "from the wood pool" here?

Reviewer: Ah, nevermind, I see below that that isn't strictly true in this change

amount transferred is limited by available carbon and nitrogen:

\begin{equation}
\mathbb{F}^C_{\text{leaf,on,}t} =

> f_{\text{leaf,on,}t} \cdot F^C_{\text{demand,leaf,on,}t}

Reviewer: Do we need the `,t` subscript? We don't use it anywhere else in this doc (I think).

Author: removed

\label{eq:leaf_on_creation}
\end{equation}

where $0 \le f_{\text{leaf,on,}t} \le 1$.

A minimal limiter combines carbon and nitrogen constraints:

\begin{equation}

> \mathbb{f}_{\text{leaf,on,}t} =

Author: please stop

\min(f^C_{\text{limit,}t}, f^N_{\text{limit,}t})
\label{eq:leaf_on_limiter}
\end{equation}

with $0 \le f^C_{\text{limit,}t} \le 1$ and $0 \le f^N_{\text{limit,}t} \le 1$.

Carbon available for leaf growth at leaf-on comes from a fraction of the perennial wood and coarse root biomass pools:

\begin{equation}
\mathbb{C}_{\text{realloc,}t} =
\mathbb{f}^C_{\text{realloc}} \cdot
(C_{\text{wood,}t} + C_{\text{coarse root,}t})
\label{eq:leaf_on_realloc_c}
\end{equation}

where $0 \le \mathbb{f}^C_{\text{realloc}} \le 1$. Leaf-on reallocation uses structural perennial biomass only. It does not draw
from $C_{\text{wood,storage}}$, because the storage pool is a bookkeeping buffer for recent carbon allocation lag rather than a carbon pool available for reallocation.

If structural biomass is greater than zero, carbon is transferred to leaf carbon. 
Carbon is taken from the source pools in proportion to their size:

\begin{equation}
\mathbb{F}^C_{\text{source,}j,t} =

> F^C_{\text{leaf,on,}t}

Author: See other comments on the use of mathbb. this is intentional

> F^C_{\text{leaf,on,}t}

Author: Respectfully: Perhaps this is just an esoteric notation that you can't wrap your head around. But please accept how I am using it and stop trying to dissuade me.

\frac{C_{j,t}}{C_{\text{wood,}t} + C_{\text{coarse root,}t}}
\label{eq:leaf_on_source_c}
\end{equation}

\begin{equation*}
\small j \in \{\text{wood, coarse root}\}
\end{equation*}

If the source biomass is zero, realized leaf-on growth is zero. If the source biomass is zero and there are no leaves,
the plant will never regrow leaf biomass.

The carbon-side limiter is:

\begin{equation}
\mathbb{f}^C_{\text{limit,}t} =

> \min\left(1, \frac{C_{\text{realloc,}t}}{F^C_{\text{demand,leaf,on,}t}}\right)

Author: this case won't happen. not worth documenting. If anything, throw an error if a leaf tries to grow 0 amount.

\label{eq:leaf_on_c_limiter}
\end{equation}

and the nitrogen-side limiter is:

\begin{equation}
\mathbb{f}^N_{\text{limit,}t} =
\min\left(1, \frac{F^N_{\text{supply,leaf,on,}t}}{F^N_{\text{demand,leaf,on,}t}}\right)
\label{eq:leaf_on_n_limiter}
\end{equation}

> where $F^N_{\text{supply,leaf,on,}t}$ includes plant N storage, mineral N uptake, and N fixation.

Author: I'd rather keep it undefined when there is no demand. This should be caught further upstream: if leafGrowth>0 and C:N<Inf, then we don't need to have any of these equations evaluated. Basically, the suggested text describes a case that shouldn't happen.

The change in plant leaf carbon $(C_\text{leaf})$ over time is given by the balance of leaf production $(L)$ and leaf
litter production $(F^C_\text{litter,leaf})$.

#### Leaf On {#leaf-on}

Leaf-on events define the timing of leaf emergence. Leaf-on requests a transfer of carbon to the leaf pool, and the
amount transferred is limited by available carbon and nitrogen:

\begin{equation}
\mathbb{F}^C_{\text{leaf,on,}t} =
f_{\text{leaf,on,}t} \cdot F^C_{\text{demand,leaf,on,}t}
\label{eq:leaf_on_creation}
\end{equation}

where $0 \le f_{\text{leaf,on,}t} \le 1$.

A minimal limiter combines carbon and nitrogen constraints:

\begin{equation}
\mathbb{f}_{\text{leaf,on,}t} =
\min(f^C_{\text{limit,}t}, f^N_{\text{limit,}t})
\label{eq:leaf_on_limiter}

> \end{equation}

Author: I just want to use \mathbb enough to indicate that the structure is proposed. It doesn't need to be everywhere. If you feel strongly, please implement

with $0 \le f^C_{\text{limit,}t} \le 1$ and $0 \le f^N_{\text{limit,}t} \le 1$.

> Leaf-on carbon is supplied by reallocatable perennial biomass from wood and coarse root pools:

Author: ```suggestion
Carbon available for leaf growth at leaf-on comes from a fraction of the perennial wood and coarse root biomass pools:
```

\begin{equation}
\mathbb{C}_{\text{realloc,}t} =
f^C_{\text{realloc}} \cdot
(C_{\text{wood,}t} + C_{\text{coarse root,}t})
\label{eq:leaf_on_realloc_c}
\end{equation}

where $0 \le f^C_{\text{realloc}} \le 1$. Leaf-on reallocation uses structural perennial biomass only. It does not draw

> from $C_{\text{wood,storage}}$, because the storage pool is a bookkeeping buffer for recent carbon allocation lag rather than a carbon pool available for reallocation.

Author: might be overexplaining

If the source biomass is greater than zero, carbon transfer from source pools to leaf carbon is mass conserving. Source

> pools are debited in proportion to structural pool size:

Author: ```suggestion
If structural biomass is greater than zero, carbon is transferred to leaf carbon. 
Carbon is taken from the source pools in proportion to their size:
```

\begin{equation}
\mathbb{F}^C_{\text{source,}j,t} =
F^C_{\text{leaf,on,}t}
\frac{C_{j,t}}{C_{\text{wood,}t} + C_{\text{coarse root,}t}}
\label{eq:leaf_on_source_c}
\end{equation}

\begin{equation*}
\small j \in \{\text{wood, coarse root}\}
\end{equation*}

> This makes the source-pool debit auditable:
> 
> \begin{equation}
> \sum_j F^C_{\text{source,}j,t} = F^C_{\text{leaf,on,}t}
> \label{eq:leaf_on_source_c_sum}
> \end{equation}
> 

Author: ```suggestion
```

If the source biomass is zero, realized leaf-on growth is zero. If the source biomass is zero and there are no leaves,
the plant will never regrow leaf biomass.

The carbon-side limiter is:

\begin{equation}
\mathbb{f}^C_{\text{limit,}t} =
\min\left(1, \frac{C_{\text{realloc,}t}}{F^C_{\text{demand,leaf,on,}t}}\right)
\label{eq:leaf_on_c_limiter}
\end{equation}

and the nitrogen-side limiter is:

\begin{equation}
\mathbb{f}^N_{\text{limit,}t} =
\min\left(1, \frac{F^N_{\text{supply,leaf,on,}t}}{F^N_{\text{demand,leaf,on,}t}}\right)
\label{eq:leaf_on_n_limiter}
\end{equation}

> where $F^N_{\text{supply,leaf,on,}t}$ includes plant N storage, mineral uptake, and fixation.

Author: ```suggestion
where $F^N_{\text{supply,leaf,on,}t}$ includes plant N storage, mineral N uptake, and N fixation.
```

#### Leaf Off {#leaf-off}

Leaf-off events define the timing of leaf senescence. A leaf-off event transfers leaf biomass carbon out of the leaf

> pool. When the `--litter-pool` model option is enabled, leaf litter enters the litter pool; otherwise, leaf litter is

Author: ```suggestion
pool. When the litter pool is enabled, leaf litter enters the litter pool; otherwise, leaf litter is
```

### Plant Nitrogen Demand  $F^{N}_{\text{demand}}$

Plant N demand is the amount of N required to support plant growth. This is calculated as the sum of carbon creation fluxes divided by their respective C:N ratios.

> For leaf-on reallocation, source-pool nitrogen contributes according to the source-pool C:N ratio. Net leaf-on N demand

Reviewer: This is true for all N demand, correct? (That is, source-pool N contributes...) We don't actually refer to the N storage in the equations below, so I don't think we need this. Or, perhaps, a more general statement (at the end?) stating that this demand can be met with the source-pool N.

Reviewer: Actually, that last line in this section: 
"Total plant N demand $F^N_{\text{demand,}total}$ is then partitioned between fixation and soil N uptake" 

could perhaps be:
"Total plant N demand $F^N_{\text{demand,}total}$ is reduced by the N storage pool, and then any remaining demand is partitioned between fixation and soil N uptake"
or some such.

Author: closing as no longer relevant since we decided that only storage N will support leafOn, not nMin or fixation

is the additional nitrogen needed to satisfy leaf stoichiometry:

\begin{equation}
\mathbb{F}^N_{\text{source,leaf,on,}t} =
\sum_j \frac{F^C_{\text{source,}j,t}}{CN_j}
\label{eq:leaf_on_source_n}
\end{equation}

\begin{equation*}
\small j \in \{\text{wood, coarse root}\}
\end{equation*}

\begin{equation}
\mathbb{F}^N_{\text{demand,leaf,on,}t} =
\max\left(0,
\frac{F^C_{\text{leaf,on,}t}}{CN_{\text{leaf}}} -

> F^N_{\text{source,leaf,on,}t}

Author: Maybe I haven't been perfectly consistent, but I've been keeping matbb to LHS.

\label{eq:irrig_evap}
\end{equation}

> 

Author: ```suggestion

### Leaf On/Leaf Off Events

Leaf-on and leaf-off may be triggered either by explicit `leafon` / `leafoff` records in `events.in` or by calculated phenology. 
Explicit leaf events specify timing only and take no additional event-file parameters. 
Calculated leaf-on timing is controlled by `gdd`, `soil-phenol` runtime options, or the `leafOnDay` parameter, depending on configuration; calculated
leaf-off timing is controlled by `leafOffDay`. 
Leaf on/off specified as events are incompatible with temperature (GDD or soil) or parameter based leaf timing.

Once triggered, leaf-on and leaf-off use the same dynamics described in [Leaf On](#leaf-on) and [Leaf Off](#leaf-off).

```

When a leaf on event occurs, an amount of carbon (specified by the `leafGrowth` parameter) is transferred from the wood
carbon pool to the leaf carbon pool. As leaf C:N is usually lower than wood C:N, the excess nitrogen
implied by the static C:N ratios is included as part of the plant nitrogen demand. If there is insufficient nitrogen
available for this lump-sum move, nitrogen limitation will occur. 

When a leaf off event occurs, a fraction of the leaf carbon (specified by the `fracLeafFall` parameter) is transferred
from the leaf carbon pool to the litter pool (or soil pool, if the litter pool is not being used). The corresponding
nitrogen (calculated from the leaf C:N ratio) is also transferred to the litter or soil nitrogen pool.

**Event parameters:**

| Parameter | Value                | Description       |
|-----------|----------------------|-------------------|
| Year      | integer              | Year              |
| Day       | integer              | Day of year       |
| Type      | `leafon` / `leafoff` | The type of event |

There are no other parameters needed for these events, as the amount of transfer is determined by the parameters
mentioned above.

> 

Reviewer 2: Not sure why this section is deleted

Author: I moved the section to line 209-306 so it is closer to the allocation dynamics.

But you are right, for consistency we should document leaf on/off as an event down here in the events section.  (and link to the docs)

Author: added a short section on leafOn/leafOff as events, and updated structure docs for clarity in https://github.com/PecanProject/sipnet/pull/332#pullrequestreview-4246864924
